massmigration/enforcement.py

```diff
@@ -21,51 +21,89 @@
 APPLIED_MIGRATIONS_CACHE = {}
 
 
-def migration_is_applied(migration_identifier):
+def get_migration_key(migration_id_str_or_tuple):
+    """
+    Get the migration key from the provided migration ID string or tuple.
+
+    Args:
+        migration_id_str_or_tuple (str or tuple): The migration ID string or tuple.
+
+    Returns:
+        str: The migration key.
+
+    Raises:
+        ValueError: If the provided migration key doesn't exists.
+    """
+    if isinstance(migration_id_str_or_tuple, (list, tuple)):
+        key = MigrationRecord.key_from_name_tuple(migration_id_str_or_tuple)
+    else:
+        key = migration_id_str_or_tuple
+
+    if key not in store.by_key:
+        raise ValueError(
+            f"Provided migration with key '{key}' not found."
+            f"Available keys are {', '.join(store.by_key.keys())}"
+        )
+
+    return key
+
+
+def migration_is_applied(migration_identifier, db_alias):
     """ Tells you whether or not the specified migration has been applied to the DB.
         Positive (True) responses are cached to avoid repeated DB queries.
     """
-    if is_valid_migration_id(migration_identifier):
-        key = migration_identifier
-    else:
-        key = MigrationRecord.key_from_name_tuple(migration_identifier)
+    migration_key = get_migration_key(migration_identifier)
+    cache_key = (get_migration_key(migration_identifier), db_alias)
     try:
-        return APPLIED_MIGRATIONS_CACHE[key]
+        return APPLIED_MIGRATIONS_CACHE[cache_key]
     except KeyError:
-        applied = MigrationRecord.objects.filter(key=key, is_applied=True).exists()
+        applied = MigrationRecord.objects.using(db_alias).filter(key=migration_key, is_applied=True).exists()
         if applied:
-            APPLIED_MIGRATIONS_CACHE[key] = True
+            APPLIED_MIGRATIONS_CACHE[cache_key] = True
             return True
     return False
 
 
-def requires_migration(migration_identifier, skip_in_tests=True):
+def requires_migration(migration_identifier, db_aliases=[], is_view=False, skip_in_tests=True):
     """ Function decorator which prevents the function being run if the specified migration is not
         applied.
     """
+
     def decorator(function):
         @wraps(function)
         def replacement(*args, **kwargs):
-            if not (skip_in_tests or in_tests()):
-                if not migration_is_applied(migration_identifier):
-                    raise RequiredMigrationNotApplied(
-                        f"Function '{function}'' requires migration {migration_identifier} which has "
-                        "not been applied."
-                    )
-            return function(*args, **kwargs)
-        return replacement
-    return decorator
+            def enforce_migration():
+                key = get_migration_key(migration_identifier)
 
+                migration = store.by_key.get(key, None)
```

Comment:
This should probably just blow up if the specified key doesn't refer to a migration file, right? Otherwise it will blow up further down with an AttributeError on `migration.get_allowed_db_aliases()`.
```suggestion
                migration = store.by_key[key]
```

Reply:
I think we'll actually never get here, `get_migration_key` will blow up if the `key` doesn't exist.
That said, it's another reason to access the key directly.